cpp/tests/strings/split_tests.cpp

```diff
@@ -0,0 +1,294 @@
+
+    cudf::strings_column_view strings_view( strings );
+    std::vector<const char*> h_expected1{ "héllo", nullptr, "a", "a", "", "ab", "", " a b ", " a  bbb   c" };
+    cudf::test::strings_column_wrapper expected1( h_expected1.begin(), h_expected1.end(),
+        thrust::make_transform_iterator( h_expected1.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<const char*> h_expected2{ nullptr, nullptr, "bc", "", "ab", "cd", nullptr, nullptr, nullptr };
+    cudf::test::strings_column_wrapper expected2( h_expected2.begin(), h_expected2.end(),
+        thrust::make_transform_iterator( h_expected2.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<const char*> h_expected3{ nullptr, nullptr, "déf", "bc", "cd", "", nullptr, nullptr, nullptr };
+    cudf::test::strings_column_wrapper expected3( h_expected3.begin(), h_expected3.end(),
+        thrust::make_transform_iterator( h_expected3.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<std::unique_ptr<cudf::column>> expected_columns;
+    expected_columns.push_back(expected1.release());
+    expected_columns.push_back(expected2.release());
+    expected_columns.push_back(expected3.release());
+    auto expected = std::make_unique<cudf::experimental::table>(std::move(expected_columns));
+
+    auto results = cudf::strings::rsplit(strings_view, cudf::string_scalar("_"));
+    EXPECT_TRUE( results->num_columns()==3 );
+    cudf::test::expect_tables_equal(*results,*expected);
+}
+
+TEST_F(StringsSplitTest, RSplitWhitespace)
+{
+    std::vector<const char*> h_strings{ "héllo", nullptr, "a_bc_déf", "", " a\tb ", " a\r bbb   c" };
+    cudf::test::strings_column_wrapper strings( h_strings.begin(), h_strings.end(),
+        thrust::make_transform_iterator( h_strings.begin(), [] (auto str) { return str!=nullptr; }));
+
+    cudf::strings_column_view strings_view( strings );
+    std::vector<const char*> h_expected1{ "héllo", nullptr, "a_bc_déf", nullptr, "a", "a" };
+    cudf::test::strings_column_wrapper expected1( h_expected1.begin(), h_expected1.end(),
+        thrust::make_transform_iterator( h_expected1.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<const char*> h_expected2{ nullptr, nullptr, nullptr, nullptr, "b", "bbb" };
+    cudf::test::strings_column_wrapper expected2( h_expected2.begin(), h_expected2.end(),
+        thrust::make_transform_iterator( h_expected2.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<const char*> h_expected3{ nullptr, nullptr, nullptr, nullptr, nullptr, "c" };
+    cudf::test::strings_column_wrapper expected3( h_expected3.begin(), h_expected3.end(),
+        thrust::make_transform_iterator( h_expected3.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<std::unique_ptr<cudf::column>> expected_columns;
+    expected_columns.push_back(expected1.release());
+    expected_columns.push_back(expected2.release());
+    expected_columns.push_back(expected3.release());
```

Comment:
Can we use an intializer list here?

Reply:
I could not get this to compile:
```
std::vector<std::unique_ptr<cudf::column>> expected_columns{expected1.release(),expected2.release(),expected3.release()};
```
Looks like the compiler is having trouble matching up the initializer list. Here is the error in case you have a suggestion:
```
/usr/include/c++/7/bits/stl_construct.h: In instantiation of ‘void std::_Construct(_T1*, _Args&& ...) [with _T1 = std::unique_ptr<cudf::column>; _Args = {const std::unique_ptr<cudf::column, std::default_delete<cudf::column> >&}]’:
/usr/include/c++/7/bits/stl_uninitialized.h:83:18:   required from ‘static _ForwardIterator std::__uninitialized_copy<_TrivialValueTypes>::__uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = const std::unique_ptr<cudf::column>*; _ForwardIterator = std::unique_ptr<cudf::column>*; bool _TrivialValueTypes = false]’
/usr/include/c++/7/bits/stl_uninitialized.h:134:15:   required from ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = const std::unique_ptr<cudf::column>*; _ForwardIterator = std::unique_ptr<cudf::column>*]’
/usr/include/c++/7/bits/stl_uninitialized.h:289:37:   required from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, std::allocator<_Tp>&) [with _InputIterator = const std::unique_ptr<cudf::column>*; _ForwardIterator = std::unique_ptr<cudf::column>*; _Tp = std::unique_ptr<cudf::column>]’
/usr/include/c++/7/bits/stl_vector.h:1331:33:   required from ‘void std::vector<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = const std::unique_ptr<cudf::column>*; _Tp = std::unique_ptr<cudf::column>; _Alloc = std::allocator<std::unique_ptr<cudf::column> >]’
/usr/include/c++/7/bits/stl_vector.h:387:21:   required from ‘std::vector<_Tp, _Alloc>::vector(std::initializer_list<_Tp>, const allocator_type&) [with _Tp = std::unique_ptr<cudf::column>; _Alloc = std::allocator<std::unique_ptr<cudf::column> >; std::vector<_Tp, _Alloc>::allocator_type = std::allocator<std::unique_ptr<cudf::column> >]’
/cudf/cpp/tests/strings/split_tests.cpp:126:124:   required from here
/usr/include/c++/7/bits/stl_construct.h:75:7: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = cudf::column; _Dp = std::default_delete<cudf::column>]’
     { ::new(static_cast<void*>(__p)) _T1(std::forward<_Args>(__args)...); }
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/7/bits/locale_conv.h:41:0,
                 from /usr/include/c++/7/locale:43,
                 from /usr/include/c++/7/iomanip:43,
                 from /conda/envs/rapids/include/rmm/detail/memory_manager.hpp:29,
                 from /conda/envs/rapids/include/rmm/rmm.hpp:26,
                 from /conda/envs/rapids/include/rmm/rmm.h:5,
                 from /cudf/cpp/include/cudf/utilities/error.hpp:8,
                 from /cudf/cpp/include/cudf/column/column_factories.hpp:20,
                 from /cudf/cpp/tests/strings/split_tests.cpp:17:
/usr/include/c++/7/bits/unique_ptr.h:388:7: note: declared here
       unique_ptr(const unique_ptr&) = delete;
       ^~~~~~~~~~
```
I tried wrapping each parm in `std::move` as well.

Reply:
I don't think initializer list here accepts move semantics which would be required when passing a `unique_ptr` type.